src/Main.elm

```diff
@@ -197,39 +242,65 @@ phraseResultToPhraseList result =
             []
 
 
+debounce : String -> Debouncer -> ( Debouncer, Cmd Msg )
+debounce value debouncer =
+    ( { debouncer | value = value, id = debouncer.id + 1 }
+    , Process.sleep debouncer.timeout
+        |> Task.perform (\_ -> FinishedWord (debouncer.id + 1))
+    )
+
+
+debouncerWaiting : Model -> Bool
+debouncerWaiting model =
+    model.word /= model.debouncer.value
+
+
 
 -- VIEW
 
 
 view : Model -> Html Msg
 view model =
-    case model.rhymes of
+    div []
+        [ inputView model
+        , punView model
+        ]
+
+
+punView : Model -> Html Msg
+punView model =
+    case model.phrases of
+        NotAsked ->
+            text "Type a word above for puns"
+
         Failure ->
             text "It failed :("
 
         Loading ->
             text "Loading..."
 
         Success _ ->
-            case model.phrases of
+            case model.rhymes of
+                NotAsked ->
+                    if debouncerWaiting model then
```

Comment:
Is an `if` statement the best here? I could also probably figure out some way to use a type like `StillTyping` and `DoneTyping` instead, but I guessed that it would be too over-engineered. Maybe I'm feeling weird about it because I don't see `if`s used much in elm and also it's an `if` inside a `case` inside a `case` and I need to extract some stages.